### PR TITLE
[Function] Implement `TO_DATE` and `TO_TIMESTAMP` function

### LINKED ISSUE #829: [Function] implement `TO_DATE` and `TO_TIMESTAMP` function
Support `to_date` and `to_timestamp` function that converts string into `DATE` and `TIMESTAMP` datatype. 

For your reference,
the following function is expressed as [STR_TO_DATE](https://www.w3schools.com/sql/trymysql.asp?filename=trysql_func_mysql_str_to_date).
Also following is an example converting string to timestamp [STR_TO_DATE ](https://www.w3schools.com/sql/trymysql.asp?filename=trysql_func_mysql_str_to_date4).

Thank you

### PR DESCRIPTION
**Pull request by ChobobDev**
Resolves #829

### CODE REVIEW

File: core/src/executor/evaluate/function.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+use super::ChronoFormatError;
```
Comment (ever0de) [inline core/src/executor/evaluate/function.rs:1]: Combine it with the import statement below

File: core/src/executor/evaluate/error.rs [hunk 2]
```diff
@@ -61,3 +64,60 @@ pub enum EvaluateError {
     #[error("format function does not support following data_type: {0}")]
     UnsupportedExprForFormatFunction(String),
 }
+
+#[derive(Error, Serialize, Debug, PartialEq)]
+pub enum ChronoFormatError {
```
Comment (ever0de) [inline core/src/executor/evaluate/error.rs:69]:
```rust

impl ChronoFormatError {
    pub fn err_into(error: chrono::format::ParseError) -> crate::result::Error {
        let error: ChronoFormatError = error.into();
        let error: EvaluateError = error.into();
        error.into()
    }
}
```

Let's implement this to improve our two-step `into` operation.

File: core/src/executor/evaluate/function.rs [hunk 3]
```diff
@@ -429,3 +431,37 @@ pub fn format(name: String, expr: Evaluated<'_>, format: Evaluated<'_>) -> Resul
...
+        Value::Str(expr) => {
+            let format = eval_to_str!(name, format);
+            match chrono::NaiveDate::parse_from_str(&expr, &format) {
+                Ok(a) => Ok(Value::Date(a)),
+                Err(error) => {
+                    let error: ChronoFormatError = error.into();
+                    let error: EvaluateError = error.into();
+                    Err(error.into())
+                }
+            }
```
Comment (ever0de) [inline core/src/executor/evaluate/function.rs:446]:
```suggestion
            chrono::NaiveDate::parse_from_str(&expr, &format)
                .map(Value::Date)
                .map_err(ChronoFormatError::err_into)
```

It gets so simple. How is it?

File: core/src/executor/evaluate/function.rs [hunk 4]
```diff
@@ -429,3 +431,37 @@ pub fn format(name: String, expr: Evaluated<'_>, format: Evaluated<'_>) -> Resul
...
+        Value::Str(expr) => {
+            let format = eval_to_str!(name, format);
+            match chrono::NaiveDateTime::parse_from_str(&expr, &format) {
+                Ok(a) => Ok(Value::Timestamp(a)),
+                Err(error) => {
+                    let error: ChronoFormatError = error.into();
+                    let error: EvaluateError = error.into();
+                    Err(error.into())
+                }
+            }
```
Comment (ever0de) [inline core/src/executor/evaluate/function.rs:463]:
```suggestion
            chrono::NaiveDateTime::parse_from_str(&expr, &format)
                .map(Value::Timestamp)
                .map_err(ChronoFormatError::err_into)
```

Same above too

File: core/src/executor/evaluate/error.rs [hunk 5]
```diff
@@ -61,3 +64,60 @@ pub enum EvaluateError {
...
+    fn from(error: chrono::format::ParseError) -> ChronoFormatError {
+        match error.kind() {
+            chrono::format::ParseErrorKind::OutOfRange => ChronoFormatError::OutOfRange,
+            chrono::format::ParseErrorKind::Impossible => ChronoFormatError::Impossible,
+            chrono::format::ParseErrorKind::NotEnough => ChronoFormatError::NotEnough,
+            chrono::format::ParseErrorKind::Invalid => ChronoFormatError::Invalid,
+            chrono::format::ParseErrorKind::TooShort => ChronoFormatError::TooShort,
+            chrono::format::ParseErrorKind::TooLong => ChronoFormatError::TooLong,
+            chrono::format::ParseErrorKind::BadFormat => ChronoFormatError::BadFormat,
+            chrono::format::ParseErrorKind::__Nonexhaustive => ChronoFormatError::Unreachable,
```
Comment (ever0de) [inline core/src/executor/evaluate/error.rs:120]: Repeated `chrono::format::ParseErrorKind` may reduce readability.

```Suggestion
           Use chrono::format::ParseErrorKind::*;

           error.kind() match {
               OutOfRange => ChronoFormatError::OutOfRange,
               Impossible => ChronoFormatError:: Impossible,
               NotEnough => ChronoFormatError::NotEnough,
               invalid => ChronoFormatError::invalid,
               TooShort => ChronoFormatError::TooShort,
               TooLong => ChronoFormatError::TooLong,
               BadFormat => ChronoFormatError::BadFormat,
               __Nonexhaustive => ChronoFormatError::unreachable,
           }
```
How about this?

File: core/src/executor/evaluate/function.rs (hunk 1 above)
Comment (ChobobDev) [reply core/src/executor/evaluate/function.rs:1]: Oh, thanks for poting this out, I will apply this ASAP

File: core/src/executor/evaluate/error.rs (hunk 5 above)
Comment (ChobobDev) [reply core/src/executor/evaluate/error.rs:120]: I totally agree with this

File: core/src/executor/evaluate/function.rs (hunk 3 above)
Comment (ChobobDev) [reply core/src/executor/evaluate/function.rs:446]: It looks way simple than before, I think it also increases readability

File: core/src/executor/evaluate/error.rs (hunk 2 above)
Comment (ever0de) [reply core/src/executor/evaluate/error.rs:69]: Well... the question may be where is the definition for this.
Even if it exists as a single function, there is no problem. If there is a suitable place, please change it there.

File: core/src/executor/evaluate/error.rs [hunk 6]
```diff
@@ -1,12 +1,16 @@
 use {
     crate::ast::{Aggregate, Expr},
+    chrono::format::ParseErrorKind::*,
```
Comment (ever0de) [inline core/src/executor/evaluate/error.rs:3]: Importing this across the `evaluate/error` module can be tricky. (e.g. when declaring or using the same name as a variant of `ParseErrorKind`)
This changes it to use only where it is used.

File: core/src/executor/evaluate/error.rs [hunk 7]
```diff
@@ -61,3 +65,68 @@ pub enum EvaluateError {
...
+impl ChronoFormatError {
+    pub fn err_into(error: chrono::format::ParseError) -> crate::result::Error {
+        let error: ChronoFormatError = error.into();
+        let error: EvaluateError = error.into();
+        error.into()
+    }
+}
+
+impl From<chrono::format::ParseError> for ChronoFormatError {
+    fn from(error: chrono::format::ParseError) -> ChronoFormatError {
```
Comment (ever0de) [inline core/src/executor/evaluate/error.rs:120]:
```suggestion
    fn from(error: chrono::format::ParseError) -> ChronoFormatError {
        use chrono::format::ParseErrorKind::*;
```

File: core/src/executor/evaluate/error.rs (hunk 6 above)
Comment (ChobobDev) [reply core/src/executor/evaluate/error.rs:3]: Oh, I will move this to where it belongs, thank you for letting me know

**Review by ever0de — APPROVED** [review]
Looks good 👍

File: core/src/executor/evaluate/error.rs [hunk 8]
```diff
@@ -7,6 +7,9 @@ use {
 
 #[derive(Error, Serialize, Debug, PartialEq)]
 pub enum EvaluateError {
+    #[error(transparent)]
+    ChronoFormat(#[from] ChronoFormatError),
```
Comment (panarch) [inline core/src/executor/evaluate/error.rs:11]: That's interesting that we have 2-level depth error conversion.
Looks good :) thanks!

**Review by panarch — APPROVED** [review]
We now have support new two helpful which handles dates.
Looks all good! Thanks a lot 👍

**Pull request merged**